solr/core/src/java/org/apache/solr/core/SolrXmlConfig.java

```diff
@@ -128,19 +129,23 @@ public static NodeConfig fromFile(SolrResourceLoader loader, Path configFile, Pr
   }
 
   /** TEST-ONLY */
-  public static NodeConfig fromString(SolrResourceLoader loader, String xml) {
-    return fromInputStream(loader, new ByteArrayInputStream(xml.getBytes(StandardCharsets.UTF_8)), new Properties());
+  public static NodeConfig fromString(Path solrHome, String xml) {
+    return fromInputStream(
+        solrHome,
+        new ByteArrayInputStream(xml.getBytes(StandardCharsets.UTF_8)),
+        new Properties());
   }
 
-  public static NodeConfig fromInputStream(SolrResourceLoader loader, InputStream is, Properties substituteProps) {
+  public static NodeConfig fromInputStream(Path solrHome, InputStream is, Properties substituteProps) {
+    SolrResourceLoader loader = new SolrResourceLoader(solrHome);
```

Comment:
Does this need to be closed?

Reply:
I looked into this a bit tonight; I didn't consider the close-ability of SRLs before.  Apparently it's because of the embedded URLClassLoader which _ought to_ be closed.  This was discussed when it was made closeable in https://issues.apache.org/jira/browse/SOLR-4791 and it seems to only affect Windows practically speaking.  The CoreContainer.shutdown method will close it's resourceLoader which is fetched from this SolrXmlConfig.  That's the particular instance you comment on here.  But I don't see that a Core's SRL will get closed.  Shrug; I dunno if I should try to address that in this PR.